fleet/comms_rpc.go

```diff
@@ -46,6 +46,7 @@ type AgentPolicyRPCPayload struct {
 	Action    string      `json:"action"`
 	ID        string      `json:"id"`
 	DatasetID string      `json:"dataset_id"`
+	GroupID   string      `json:"group_id"`
```

Comment:
So this needs to be a string[] (array) because during NotifyAgentAllDatasets we will need to fill this out too, and a policy could be coming from multiple groups

Reply:
got it, but that doesn't mean that we should also fill DatasetID as an array too? since a policy can also be linked to many datasets